apm-agent-core/src/main/java/co/elastic/apm/configuration/CoreConfiguration.java

```diff
@@ -104,6 +110,52 @@ public void assertValid(Double value) {
         .dynamic(true)
         .buildWithDefault(500);
 
+    private final ConfigurationOption<List<WildcardMatcher>> sanitizeFieldNames = ConfigurationOption
+        .builder(new ListValueConverter<>(new WildcardMatcherValueConverter()), List.class)
+        .key("sanitize_field_names")
+        .configurationCategory(CORE_CATEGORY)
+        .description("Sometimes it is necessary to sanitize the data sent to Elastic APM,\n" +
+            "e.g. remove sensitive data.\n" +
+            "\n" +
+            "Configure a list of wildcard patterns of field names which should be sanitized.\n" +
+            "These apply for example to HTTP headers and `application/x-www-form-urlencoded` data.\n" +
+            "\n" +
+            "NOTE: Data in the query string is considered non-sensitive,\n" +
+            "as sensitive information should not be sent in the query string.\n" +
+            "See https://www.owasp.org/index.php/Information_exposure_through_query_strings_in_url for more information\n" +
+            "\n" +
+            "NOTE: Review the data captured by Elastic APM carefully to make sure it does not capture sensitive information.\n" +
+            "If you do find sensitive data in the Elasticsearch index,\n" +
+            "you should add an additional entry to this list (make sure to also include the default entries)."
+            /* A disadvantage of this approach is when a user adds a custom value,
+             * they don't automatically pick up new default values.
+             * But the possibility to remove default values which are leading to false positive for the user
+             * outweights this disadvantage imho.
+             * Another advantage of having a configuration option vs offering a Filter or Processor in the public API is
+             * that we don't have to expose the internal data format to the public API.
+             */
+        )
+        .dynamic(true)
+        .tags("security")
+        .buildWithDefault(Arrays.asList(
+            WildcardMatcher.valueOf("(?i)password"),
+            WildcardMatcher.valueOf("(?i)passwd"),
+            WildcardMatcher.valueOf("(?i)pwd"),
+            WildcardMatcher.valueOf("(?i)secret"),
+            WildcardMatcher.valueOf("(?i)token"),
+            WildcardMatcher.valueOf("(?i)*key"),
+            WildcardMatcher.valueOf("(?i)*token"),
+            WildcardMatcher.valueOf("(?i)*session*"),
+            WildcardMatcher.valueOf("(?i)*credit*"),
+            WildcardMatcher.valueOf("(?i)*card*"),
+            // HTTP request header for basic auth, contains passwords
+            WildcardMatcher.valueOf("(?i)authorization"),
+            // HTTP request header cookies are already captured in Request.cookies
+            WildcardMatcher.valueOf("(?i)cookie"),
```

Comment:
Does this mean that the `cookie` header will be set to `[REDACTED]` on every request? If that's the case, I'd rather remove it, as the value is close to zero (only thing it tells you that the request did in fact have a cookie header), and the potential for confusion is high.

In the Python agent, we handle the `cookie` header separately, by turning it into a map, sanitize the map, then turn it back into a string: https://github.com/elastic/apm-agent-python/blob/e74bd1a55b275e4d147699308c22785595c4636e/elasticapm/processors.py#L93-L96

Reply:
It also tells the user that if they do want to collect these headers, they have to change the configuration. When I'm just dropping the header, they might not know what's going on.

But it probably makes sense to just hard-code to leave out the `Cooke` header as there is also `Request.cookie` and to leave the sanitization for `(?i)set-cookie` and `(?i)authorization` as-is.

Oh, and I also forgot to document that `(?i)` means case insensitive matching.

Reply:
I'd like to avoid having to parse the cookie string to a map and back to a string as this creates a lot of garbage, which I'm trying to avoid in the Java agent.